sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -168,3 +187,44 @@ class Reduce(WithInput, Generic[MeasurementUnit, InputType, OutputType]):
     aggregate_backend: Optional[AggregationBackend[OutputType]] = None
     group_by_key: Optional[GroupBy] = None
     step_type: StepType = StepType.REDUCE
+
+
+@dataclass
+class Batch(Reduce, Generic[MeasurementUnit, InputType]):
+    """
+    A step batch up the results of the prior step.
+
+    Batch can be configured via batch size, which can be
+    an event time duration or a count of events.
+    """
+
+    # TODO: Use the idea of custom triggers to close window
+    # by either size or time
+    batch_size: MeasurementUnit
+    step_type: StepType = StepType.REDUCE
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        self.windowing: TumblingWindow[MeasurementUnit] = TumblingWindow(self.batch_size)
+        self.aggregate_fn = BatchBuilder[InputType]
+
+
+@dataclass
+class FlatMapStep(WithInput):
```

Comment:
Open to different namings

Reply:
`OneToManyTransform` ?
`Unpack` 
`Unfold`